web/src/components/itemList.js

```diff
@@ -24,25 +24,23 @@ class ItemList extends Component {
         this.state = {
             items: items,
             currency: currency,
+            selected: selected,
         };
-        this.selected = selected;
         this.rate = 0.001;
     }
 
     itemAdd = event => {
         const idx = event.target.getAttribute('id');
         this.props.addItems(this.state.items[idx].price);
-        this.selected[idx] += 1
+        this.state.selected[idx] += 1
```

Comment:
This is the wrong way to update state.
The [React docs](https://reactjs.org/docs/react-component.html) says:

> Treat this.state as if it were immutable.

same problems on line 43